tests/server/api/test_work_queues.py

```diff
@@ -495,6 +502,60 @@ async def test_read_work_queue_runs_associated_deployments_return_status_of_read
         assert updated_deployment_response.status_code == status.HTTP_200_OK
         assert updated_deployment_response.json()["status"] == "READY"
 
+    async def test_read_work_queue_runs_updates_work_queue_status(
+        self,
+        client,
+        work_queue,
+        session,
+    ):
+        # Verify the work queue is initially not ready
+        wq_response = await client.get(f"/work_queues/{work_queue.id}")
+        assert wq_response.status_code == status.HTTP_200_OK
+        assert wq_response.json()["status"] == "NOT_READY"
+
+        # Trigger a polling operation
+        response = await client.post(
+            f"/work_queues/{work_queue.id}/get_runs",
+            json=dict(agent_id=str(uuid.uuid4())),
+        )
+        assert response.status_code == status.HTTP_200_OK
+
+        # Verify the work queue is now ready
+        wq_response = await client.get(f"/work_queues/{work_queue.id}")
+        assert wq_response.status_code == status.HTTP_200_OK
+        assert wq_response.json()["status"] == "READY"
+
+    async def test_read_work_queue_runs_does_not_update_a_paused_work_queues_status(
+        self,
+        client,
+        work_queue,
+        session,
+    ):
+        # Move the queue into a PAUSED state
+        new_data = WorkQueueUpdate(is_paused=True).dict(
+            json_compatible=True, exclude_unset=True
+        )
+        response = await client.patch(f"/work_queues/{work_queue.id}", json=new_data)
+        assert response.status_code == status.HTTP_204_NO_CONTENT
+
+        # Verify the work queue is PAUSED
+        wq_response = await client.get(f"/work_queues/{work_queue.id}")
+        assert wq_response.status_code == status.HTTP_200_OK
+        assert wq_response.json()["status"] == "PAUSED"
+        assert wq_response.json()["is_paused"] is True
+
+        # Trigger a polling operation
+        response = await client.post(
+            f"/work_queues/{work_queue.id}/get_runs",
+            json=dict(agent_id=str(uuid.uuid4())),
+        )
+        assert response.status_code == status.HTTP_200_OK
+
+        # Verify the work queue status is still PAUSED
+        wq_response = await client.get(f"/work_queues/{work_queue.id}")
+        assert wq_response.status_code == status.HTTP_200_OK
+        assert wq_response.json()["status"] == "PAUSED"
+
```

Comment:
awesome "regression test" kinda

Reply:
If only we'd had this test on our other PR :)